wled00/colors.cpp

```diff
@@ -91,6 +91,113 @@ void setRandomColor(byte* rgb)
   colorHStoRGB(lastRandomIndex*256,255,rgb);
 }
 
+/*  
+ *generates a random palette based on color theory
+ */
+
+CRGBPalette16 generateRandomPalette(CRGBPalette16 &basepalette) 
+{
+  CHSV palettecolors[4]; //array of colors for the new palette
+  uint8_t keepcolorposition = random8(4); //color position of current random palette to keep
+  palettecolors[keepcolorposition] = rgb2hsv_approximate(basepalette.entries[keepcolorposition*5]); //read one of the base colors of the current palette
```

Comment:
it is. if(i==keepcolorposition) continue; //skip the base color 
see line 182

Reply:
what about other swaps? what's their use then?

Reply:
to make the palette more random as the colors are picked from the color wheel in an ordered fashion, you would never get a greenish color next to a blueish one if picking 4 colors for example. it would always look 'rainbowy', which is not what I was going for with a random generator. you can try to remove the hue shuffeling and see if it is more to your taste?

Reply:
> it would always look 'rainbowy', which is not what I was going for

Oh! And my expectation was exactly the "rainbowy" type of palette. That clarifies it.
What if you'd implement a random decision about non-"rainbowy" palette? So that sometimes you'd get *smooth* palette and sometimes not?

Reply:
sure, right now it would happen quite rarely. I will make it 50:50, shuffle or no shuffle and see what I get.
I was also thinking of adding a 10% chance of desaturating all colors more for a pastel color look.